src/components/model/Work/components/URLInput/hooks/index.ts

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react';
+import type { KeyboardEvent } from 'react';
+
+type IUseURLInput = {
+  links: string[];
+  onInputKeyDown: (event: KeyboardEvent<HTMLInputElement>) => void;
+  deleteLink: (link: string) => void;
+  invalidUrlWarning: boolean;
+  duplicateUrlWarning: boolean;
+};
+
+const isValidUrl = (url: string): boolean =>
+  url.match(/^https?:\/\/[\w/:%#&?()~.=+-]+$/) !== null;
+
+export const useURLInput = (): IUseURLInput => {
+  const [invalidUrlWarning, setInvalidUrlWarning] = useState<boolean>(false);
+  const [duplicateUrlWarning, setDuplicateUrlWarning] =
+    useState<boolean>(false);
```

Comment:
```suggestion
type WarningStatus = "ok"|"invalidUrl"|"duplicateUrl"
type WarningMessage =warningMessage[keyof warningMessage]
type Warning = {
status:WarningStatus
message:WarningMessage
}
const warningMessage= {
"ok":"",
"invalidUrl":"url is invalid",
"duplicateUrl": "url is already exist"
} as const 
  const [warning, setWarning] = useState<Warning>({status:"ok",message:""});
```

ってやると状態持つ変数が一つで済むからいいかも？
(型は記憶で書いたので間違ってたらすみません)